### PR TITLE
Fix is-int and is-float; add tests

### LINKED ISSUE #146: Fix `is-int` and `is-float` attributes of `number` using `as-bytes`
We are planning to make `eq` attribute to be an atom https://github.com/objectionary/eo/issues/2547.
Attributes `is-float` and `is-int` of object `number` assumes, that `0.0` is not equals to `-0.0`. After closing https://github.com/objectionary/eo/issues/2547 this won't be true. Thus, we need to move logic of  `0.0` and `-0.0` comparison from current `eq` attribute to `is-float` and `is-int` attributes, and compare this values "`as-bytes`".

### PR DESCRIPTION
**Pull request by c71n93**
Closes #146

---

### Focus of this PR:
This PR focuses on adding new methods to the `number` class in the `math` package. These methods check if a number is equal to zero, whether it is an integer or a float, and if it is NaN.

### Detailed summary:
- Added a new method `is-int-zero` to check if a number is an integer zero.
- Added a new method `is-float-zero` to check if a number is a floating-point zero.
- Added a new method `is-nan` to check if a number is NaN.
- Added test cases to verify the functionality of the new methods.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by c71n93** [comment]
@maxonfjvipon could you check this one, please?

File: src/main/eo/org/eolang/math/number.eo [hunk 1]
```diff
@@ -55,17 +55,38 @@
...
-        value.eq 0
+        ^.is-int-zero
         FALSE
         eq.
           div.
             value
             value
           1.0
 
+  # Checking if number is integer 0
```
Comment (maxonfjvipon) [inline src/main/eo/org/eolang/math/number.eo:69]: @c71n93 [suggestion]: maybe move these methods inside `is-float` and `is-int` objects? I think there's no need to have them in `number` (at least for now)

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]
@c71n93 Good one! Just a several comments from my side

File: src/main/eo/org/eolang/math/number.eo [hunk 2]
```diff
@@ -55,17 +55,38 @@
...
+      value.as-bytes
+      value.neg.as-bytes
+
+  # Checking if number is floatint point 0.0 or -0.0
+  [] > is-float-zero
+    ^.n > value!
+    if. > @
+      eq.
+        value.as-bytes
+        0.0.as-bytes
```
[the remark this answers is not in the document]
Comment (maxonfjvipon) [reply src/main/eo/org/eolang/math/number.eo:82]: @c71n93 or at least you can cache `0.0.as-bytes`

[the remark this answers is not in the document]
Comment (c71n93) [reply src/main/eo/org/eolang/math/number.eo]: @maxonfjvipon done

**Review by maxonfjvipon — APPROVED** [review]

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**